packages/wallets/glow/src/adapter.ts

```diff
@@ -0,0 +1,270 @@
+
+    private _connecting: boolean;
+    private _wallet: GlowWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(_config: GlowWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            const handler = (event: MessageEvent<any>) => {
+                if (typeof event.data === 'object' && event.data.__glow_loaded) {
+                    if (this._readyState !== WalletReadyState.Installed) {
+                        this._readyState = WalletReadyState.Installed;
+                        this.emit('readyStateChange', this._readyState);
+                    }
+                    window.removeEventListener('message', handler);
+                }
+            };
+
+            window.addEventListener('message', handler);
+
+            scopePollingDetectionStrategy(() => {
+                if (window.solana?.isGlow) {
+                    if (this._readyState !== WalletReadyState.Installed) {
+                        this._readyState = WalletReadyState.Installed;
+                        this.emit('readyStateChange', this._readyState);
+                    }
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.isConnected;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) {
+                return;
+            }
+            if (this._readyState !== WalletReadyState.Installed) {
+                throw new WalletNotReadyError();
+            }
```

Comment:
style nit: project code style is to not use braces on early return or throw.

I wonder if there's a lint fix for this...

Reply:
I would recommend just using [Prettier](https://prettier.io/) and then not worrying about any style stuff. That's what we do for the most part and it tends to work.